Spigot-Server-Patches/0592-Zero-pad-chunks-before-saving-them-to-the-disk.patch

```diff
@@ -0,0 +1,47 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Toon Schoenmakers <nighteyes1993@gmail.com>
+Date: Mon, 12 Oct 2020 18:59:12 +0200
+Subject: [PATCH] Zero pad chunks before saving them to the disk
+
+This makes the unused data inside region files more repeatable.
+Which will result in better compression in case the entire world directory
+is compressed (like in backups). Which will naturally result in smaller backups.
+
+diff --git a/src/main/java/net/minecraft/server/RegionFile.java b/src/main/java/net/minecraft/server/RegionFile.java
+index 1751fb6934d9242e475c1a44b2a4a1ade6987766..397d35ecb3371f6d05af526cc7693a65a75fe968 100644
+--- a/src/main/java/net/minecraft/server/RegionFile.java
++++ b/src/main/java/net/minecraft/server/RegionFile.java
+@@ -513,13 +513,14 @@ public class RegionFile implements AutoCloseable {
+         }
+ 
+     }
++    private static final ByteBuffer zero = ByteBuffer.allocateDirect(4096);
+     // Paper end
+     class ChunkBuffer extends ByteArrayOutputStream {
+ 
+         private final ChunkCoordIntPair b;
+ 
+         public ChunkBuffer(ChunkCoordIntPair chunkcoordintpair) {
+-            super(8096);
++            super(8192); // Paper
+             super.write(0);
+             super.write(0);
+             super.write(0);
+@@ -532,6 +533,17 @@ public class RegionFile implements AutoCloseable {
+             ByteBuffer bytebuffer = ByteBuffer.wrap(this.buf, 0, this.count);
+ 
+             bytebuffer.putInt(0, this.count - 5 + 1);
++
++            // Paper start
++            int size = ((this.count / 4096) + 1) * 4096;
```

Comment:
If `count` is a multiple of 4096, this yields an incorrect value (+4096 of what's desired).
We need a "`Math.ceilDiv`", which doesn't exist. However, you can prove ceilDiv(m, n)=-floorDiv(-m, n). Maybe Mojang already has a util for this?

Alternatively, albeit slower, you can use `((int) Math.ceil(count / 4096D)) * 4096`

Reply:
Seems like I incorrectly copied that from one of the other functions. Update incoming where I just use said function instead.

Reply:
int size = (this.count + 4095) & ~4095